core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java

```diff
@@ -0,0 +1,1030 @@
+            }
+        }
+
+        protected boolean _isPerThread(Class<? extends Annotation> scope) {
+            return RequestScoped.class.equals(scope) || PerThread.class.equals(scope);
+        }
+
+        protected X _getInstance(InstanceBinding<X> instanceBinding) {
+            return instanceBinding.getService();
+        }
+
+        protected X _create(SupplierInstanceBinding<X> binding) {
+            Supplier<X> supplier = binding.getSupplier();
+            X t = registerDisposableSupplierAndGet(supplier);
+            if (Singleton.class.equals(binding.getScope())) {
+                _addInstance(t, binding);
+            } else if (_isPerThread(binding.getScope())) {
+                _addThreadInstance(t, binding);
+            }
+            return t;
+        }
+
+        X create() {
+            _checkUnique();
+            if (!instanceBindings.isEmpty()) {
+                return _getInstance(instanceBindings.get(0));
+            } else if (!supplierInstanceBindings.isEmpty()) {
+                return _create(supplierInstanceBindings.get(0));
+            } else if (!classBindings.isEmpty()) {
+                return _createAndStore(classBindings.get(0));
+            } else if (!supplierClassBindings.isEmpty()) {
+                return _create(supplierClassBindings.get(0));
+            }
+
+            throw new IllegalStateException(LocalizationMessages.NONINJECT_NO_BINDING(type));
+        }
+
+        protected X getInstance() {
+            X instance = instances.getInstance(type, instancesQualifiers);
+            if (instance != null) {
+                return instance;
+            }
+            return create();
+        }
+
+        List<X> allInstances() {
+            List<X> list = new LinkedList<>();
+            List<InstanceContext<?>> instanceContextList;
+
+            instanceContextList = instances.getContexts(type, instancesQualifiers);
+            if (instanceContextList != null) {
+                instanceContextList.forEach(instanceContext -> filterBinding(instanceContext.getBinding()));
+                instanceContextList.forEach(instanceContext -> list.add((X) instanceContext.getInstance()));
+            }
+
+            list.addAll(instanceBindings.stream()
+                    .map(this::_getInstance)
+                    .collect(Collectors.toList()));
+
+            list.addAll(classBindings.stream()
+                    .map(this::_createAndStore)
+                    .collect(Collectors.toList()));
+
+            list.addAll(supplierInstanceBindings.stream()
+                    .map(this::_create)
+                    .collect(Collectors.toList()));
+
+            list.addAll(supplierClassBindings.stream()
+                    .map(this::_create)
+                    .collect(Collectors.toList()));
+
+            return list;
+        }
+
+        protected abstract X _create(SupplierClassBinding<X> binding);
+
+        protected abstract X _createAndStore(ClassBinding<X> binding);
```

Comment:
I thought _ was for "do not overwrite" methods

Reply:
Methods with "_" are internal to its hierarchy, and not to be called from outside it.